internal/docstore/dynamodocstore/dynamo.go

```diff
@@ -239,62 +253,31 @@ func (c *collection) runActionsOrdered(ctx context.Context, actions []*driver.Ac
 // splitActions doesn't change the order of the input slice.
 func (c *collection) splitActions(actions []*driver.Action) [][]*driver.Action {
 	var (
-		groups [][]*driver.Action              // the actions, split; the return value
-		cur    []*driver.Action                // the group currently being constructed
-		wm     = make(map[[2]interface{}]bool) // writes group cannot contain duplicate items
+		groups [][]*driver.Action // the actions, split; the return value
+		cur    []*driver.Action   // the group currently being constructed
 	)
 	collect := func() { // called when the current group is known to be finished
 		if len(cur) > 0 {
 			groups = append(groups, cur)
 			cur = nil
-			wm = make(map[[2]interface{}]bool)
 		}
 	}
 	for _, a := range actions {
-		if len(cur) > 0 && c.shouldSplit(cur[len(cur)-1], a, wm) ||
+		if len(cur) > 0 && c.shouldSplit(cur[len(cur)-1], a) ||
 			len(cur) >= 10 { // each transaction can run up to 10 operations.
 			collect()
 		}
 		cur = append(cur, a)
-		if a.Kind != driver.Get {
-			if keys := c.primaryKey(a); keys[0] != nil {
-				wm[keys] = true
-			}
-		}
 	}
 	collect()
 	return groups
 }
 
-func (c *collection) shouldSplit(curr, next *driver.Action, wm map[[2]interface{}]bool) bool {
+func (c *collection) shouldSplit(curr, next *driver.Action) bool {
```

Comment:
Add to TODO to group actions as many as possible.

Reply:
This is all going to be obsolete soon anyway.